internal/cloud/backend_runTasks.go

```diff
@@ -138,12 +181,150 @@ func (b *Cloud) runTasksWithTaskResults(context *IntegrationContext, output Inte
+func (b *Cloud) runTaskStage(ctx *IntegrationContext, output IntegrationOutputWriter, stageID string) error {
+	err1 := b.runTasksWithTaskResults(ctx, output, func(b *Cloud, stopCtx context.Context) (*tfe.TaskStage, error) {
 		options := tfe.TaskStageReadOptions{
-			Include: []tfe.TaskStageIncludeOpt{tfe.TaskStageTaskResults},
+			Include: []tfe.TaskStageIncludeOpt{tfe.TaskStageTaskResults, tfe.PolicyEvaluationsTaskResults},
 		}
 
 		return b.client.TaskStages.Read(ctx.StopContext, stageID, &options)
 	})
+	// call to runTaskwithPolicyEvaluation
+	err2 := b.runTasksWithPolicyEvaluation(ctx, output, func(b *Cloud, stopCtx context.Context) (*tfe.TaskStage, error) {
+		options := tfe.TaskStageReadOptions{
+			Include: []tfe.TaskStageIncludeOpt{tfe.TaskStageTaskResults, tfe.PolicyEvaluationsTaskResults},
+		}
+
+		return b.client.TaskStages.Read(ctx.StopContext, stageID, &options)
+	})
+	//TODO: clean this up
+	if err1 != nil && err2 != nil {
+		return fmt.Errorf("error running task stage: %v %v", err1, err2)
+	} else if err1 != nil {
+		return err1
+	} else if err2 != nil {
+		return err2
+	}
+	return nil
+
+}
+
+func (b *Cloud) runTasksWithPolicyEvaluation(context *IntegrationContext, output IntegrationOutputWriter, fetchTaskStage taskStageReadFunc) error {
+	return context.Poll(func(i int) (bool, error) {
+		stage, err := fetchTaskStage(b, context.StopContext)
+
+		if err != nil {
+			return false, generalError("Failed to retrieve task stage", err)
+		}
+
+		if len(stage.PolicyEvaluations) == 0 {
+			return false, nil
+		}
+
+		summary := summarizePolicyEvaluationResults(stage.PolicyEvaluations)
+		if summary.unreachable {
+			output.Output("Skipping policy evaluation.")
+			output.End()
+			return false, nil
+		}
+
+		if summary.pending > 0 {
+			pendingMessage := "Evaluating ... "
+
+			if i%4 == 0 {
+				if i > 0 {
+					output.OutputElapsed(pendingMessage, len(pendingMessage))
+				}
+			}
+			return true, nil
+		}
+
+		// No more policy evaluations pending/running. Print all the results.
+		output.Output("\n------------------------------------------------------------------------\n")
+		output.Output("[bold]OPA Policy Evaluation\n")
+
+		var result, message string
+		// Currently only one policy evaluation supported : OPA
+		if stage.PolicyEvaluations[0].Status == tfe.PolicyEvaluationPassed {
+			message = "[dim] This result means that all OPA policies passed and the protected behaviour is allowed"
+			result = fmt.Sprintf("[green]%s", strings.ToUpper(string(tfe.PolicyEvaluationPassed)))
+			if stage.PolicyEvaluations[0].ResultCount.AdvisoryFailed > 0 {
+				result += " (with advisory)"
+			}
+		} else {
+			message = "[dim] This result means that one or more OPA policies failed. More than likely, this was due to the discovery of violations by the main rule and other sub rules"
+			result = fmt.Sprintf("[red]%s", strings.ToUpper(string(tfe.PolicyEvaluationFailed)))
+		}
```

Comment:
Hrmmm not sure I like assuming only 1 policy eval.  Seems dangerous as the API does say it's zero or more.